theme/views.py

```diff
@@ -30,11 +30,21 @@ def get_context_data(self, **kwargs):
             except:
                 u = User.objects.get(username=self.request.GET['user'])
 
-        res = BaseResource.objects.filter(user=u)
+        # get all resources the profile user has created
+        resources = BaseResource.objects.filter(user=u)
+
+        # if requesting user is not the profile user, then show only resources that the requesting user has access
+        if self.request.user != u:
+            if self.request.user.is_authenticated():
+                # filter out any resources the requesting user doesn't have view permission
+                resources = [res for res in resources if self.request.user.uaccess.can_view_resource(res)]
+            else:
+                # for anonymous requesting user show only resources that are either public or discoverable
+                resources = [res for res in resources if res.raccess.public or res.raccess.discoverable]
```

Comment:
@alvacouch do you also agree this line https://github.com/hydroshare/hydroshare/blob/develop/hs_access_control/models.py#L1203 should be changed to "if access_resource.public or access_resource.discoverable:" ? If so, an issue needs to be created to get it fixed so that discoverable resources will be accounted for by can_view_resource(). I think it is better to get the fix in sooner rather than later to save us troubles down the road now that discoverable is exposed to users in the interface for release 1.6.